### PR TITLE
feat: asynchrounousifying the setup function in cog

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary

i had asynchrounousified the setup function in cog

code used for testing:
testcog.py
```python
from discord.ext import commands
from discord.ext.commands import Cog

class testCog(Cog):
    def __init__(self, bot):
        self.bot = bot
    
    @commands.command()
    async def test(self, ctx):
        await ctx.send("Test")

async def setup(bot):
    bot.add_cog(testCog(bot))
    print("test done")
```

main.py
```python
import random

import discord
import asyncio
from discord.ext import commands

description = """An example bot to showcase the discord.ext.commands extension
module.
There are a number of utility commands being showcased here."""

intents = discord.Intents.default()
intents.members = True
intents.message_content = True


bot = commands.Bot(command_prefix=")", description=description, intents=intents)


@bot.event
async def on_ready():
    print(f"Logged in as {bot.user} (ID: {bot.user.id})")
    print("------")

async def setup():
    bot.load_extension("testcog")

asyncio.run(setup())
bot.run("token")
```

command used to run: `python3 main.py`
output:
```
test done
Logged in as my bot name#my bot discrim (ID: my bot id)
------
```
## Checklist

- [x] If code changes were made then they have been tested. 
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/cog.py [hunk 1]
```diff
@@ -693,7 +694,11 @@ def _load_from_module_spec(self, spec: importlib.machinery.ModuleSpec, key: str)
             raise errors.NoEntryPointError(key)
 
         try:
-            setup(self)
+            if inspect.iscoroutinefunction(setup):
+                create_task(setup(self))
```
Comment (EmreTech) [inline discord/cog.py:698]: Aren't you supposed to store the created task then eventually await for it?

File: discord/cog.py [hunk 2]
```diff
@@ -41,6 +41,7 @@
     Type,
     TypeVar,
 )
+from asyncio import create_task
```
Comment (EmreTech) [inline discord/cog.py:44]: I think it's better to just import the `asyncio` module instead of directly importing one thing from the `asyncio` module.

**Review by EmreTech — CHANGES_REQUESTED** [review]
Here are some suggestions. But about this PR in general, I don't see why it's worth making the `setup` function asynchronous.

File: discord/cog.py (hunk 1 above)
Comment (MaskDuck) [reply discord/cog.py:698]: pretty sure not.

image

**Comment by MaskDuck** [comment]
<discussed on discord>

**Pull request closed without merging**